config/crd/bases/ps.percona.com_perconaservermysqls.yaml

```diff
@@ -6694,6 +6694,13 @@ spec:
                     type: string
                 type: object
             type: object
+            x-kubernetes-validations:
+            - message: HAProxy can not be enabled with 'group-replication' clusterType
+              rule: 'self.mysql.clusterType==''group-replication'' ? !self.proxy.haproxy.enabled
+                : true'
+            - message: HAProxy and Orchestrator must be enabled with 'async' clusterType
```

Comment:
It is not true, you can have only 1 MySQL server and you should not have orc and HaProxy for it

Reply:
Ok.. then the rules would be
- HAProxy and Orchestrator must be enabled with 'async' clusterType if `mysql.size > 1`
- HAProxy must be enabled with 'async' clusterType if `mysql.size == 1`

Reply:
@inelpandzic let's remove
`HAProxy must be enabled with 'async' clusterType if mysql.size == 1`